internal/statestore/redis.go

```diff
@@ -439,6 +439,63 @@ func (rb *redisBackend) FilterTickets(ctx context.Context, filters []*pb.Filter,
 	return nil
 }
 
+// UpdateAssignments update the match assignments for the input ticket ids
+func (rb *redisBackend) UpdateAssignments(ctx context.Context, ids []string, assignment *pb.Assignment) error {
+	redisConn, err := rb.connect(ctx)
+	if err != nil {
+		return err
+	}
+	defer handleConnectionClose(&redisConn)
+
+	connectionTable := "assignment_connection"
+	propertiesTable := "assignment_properties"
+	errorsTable := "assignment_errors"
+
+	for _, id := range ids {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		default:
+			err = redisConn.Send("MULTI")
+			if err != nil {
+				redisLogger.WithError(err).Error("failed to open a multi transaction block in redis")
+				return status.Errorf(codes.Internal, "%v", err)
+			}
+
+			// Store assignment data by fields
+			err = redisConn.Send("HSET", connectionTable, id, assignment.Connection)
+			if err != nil {
+				redisLogger.WithError(err).Errorf("failed to hset assignment connection for ticket %#v", id)
```

Comment:
Error should only be emitted like this if it's a serious internal error. Like something is misconfigured. A bad query shouldn't write an error to the logs.